### ISSUE TITLE
[NETSDKE2E]Libicu package dependency issue on Ubuntu24.04

### ISSUE THREAD

**Issue by NicoleWang001** [dotnet/sdk#40506]
### Describe the bug
Due to Ubuntu having libicu74, installing .NET May update builds on Ubuntu 24.04 will return dependency problems as libicu is not installed. 
### To Reproduce

1. Try to install .NET 6.0/7.0/8.0 May update SDK via .deb on Ubuntu 24.04
### Exceptions (if any)

It will return dependency problems as libicu is not installed. 
Taking .NET 6.0.422 for example
![image](https://github.com/dotnet/sdk/assets/1578239/06f9c4f7-d22a-404d-8bda-503ff3b37dfc)

**Comment by baronfel** [comment]
@jkoritzinsky since I saw your handle on the [Runtime installer dependency list](https://github.com/dotnet/runtime/blob/f657459727d11e6559f0ff31d501b575bd50e26c/src/installer/pkg/sfx/installers/dotnet-runtime-deps/dotnet-runtime-deps-debian.proj#L9) - do you know if we support .NET 6 on ubuntu 24.04? If so, do we need to extend this ICU version list to include 74?

**Comment by nagilson** [comment]
For the record, the vendor states that this impacts 7 and 8 as well, not just 6.

**Comment by baronfel** [comment]
Yeah, that makes sense - the file is the same on 7 and 8 release branches as well.

**Comment by jkoritzinsky** [comment]
According to dotnet/core we do. I'll update the packages.

.NET 7 is out of support, so I'll only be updating .NET 6 and 8 (as we don't have a mechanism to ship .NET 7 any more).

**Comment by richlander** [comment]
Are the packages coming from packages.microsoft.com (PMC)? We don't support any PMC packages on Ubuntu 24.04.

More context:

- https://github.com/dotnet/core/discussions/9258
- https://devblogs.microsoft.com/dotnet/whats-new-for-dotnet-in-ubuntu-2404/

**Comment by jkoritzinsky** [comment]
Do we have a mechanism we can use to not provide these packages to Ubuntu 24.04 users from PMC?

Right now, I'd consider this a rough UX hole for the following scenario:

User has an Ubuntu 22.04 (or older) system, referencing PMC. They install .NET from PMC instead of from Ubuntu (maybe they wanted a newer version, for example). They upgrade to Ubuntu 24.04. They don't re-install the PMC feeds, they leave them as-is (which is wrong, but really easy to miss). They try to update dotnet, and it fails because it tries to pull updates for the dotnet-runtime-deps package, which is uninstallable on Ubuntu 24.04.

This would also be problem if they were installing the packages from PMC for Ubuntu 24.04, but I don't actually see any .NET packages in that feed (other than `dotnet-apphost-pack-6.0` for some reason).

**Comment by richlander** [comment]
Good points. We probably need to update the docs to explain how to get to a clean state with dotnet packages when upgrading to 22.04.

@adegeo

**Comment by richlander** [comment]
Zooming out. There are really two issues here:

- Why is our testing approach running into this problem?
- Improving guidance for everyone.

I would be good to get clarity on the first once, specifically why the test instructions are leading to this outcome. I also seem to remember discussing this on mail a few months ago (which would align with when this issue was created).

**Comment by adegeo** [comment]
Yeah, we should probably have something regarding this scenario in the docs, once we figure out the exact steps and versions that  need updates.

**Comment by jkoritzinsky** [comment]
I've opened https://github.com/dotnet/runtime/issues/105046 to track possible improvements here.

**Comment by ChenhuiYuan01** [comment]
This issue does not repro on SDK 9.0 and September update's SDK 6.0.

**Comment by NicoleWang001** [comment]
This issue does not repro on .NET 8..402 either. So close it.

**Comment by V0ldek** [comment]
This now happens to me with .NET 10 on Ubuntu 24.04:

```
The following packages have unmet dependencies:
 dotnet-runtime-deps-10.0 : Depends: libicu78 but it is not installable or
                                     libicu77 but it is not installable or
                                     libicu76 but it is not installable or
                                     libicu72 but it is not installable
```

The only `libicu` available in 24.04 is libicu74

```
libicu74/noble-updates,now 74.2-1ubuntu3.1 amd64 [installed,automatic]
```
